In [1]:
import pandas as pd
import matplotlib.pyplot as plt


In [2]:
vine_df = pd.read_csv('vine_table.csv')
vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
0  R1KKOXHNI8MSXU          4.0            0.0          0.0    N   
1  R26SP2OPDK4HT7          5.0            1.0          2.0    N   
2   RWQEDYAX373I1          5.0            0.0          0.0    N   
3  R231YI7R4GPF6J          5.0            0.0          0.0    N   
4  R3KO3W45DD0L1K          5.0            0.0          0.0    N   

  verified_purchase  
0                 Y  
1                 Y  
2                 Y  
3                 Y  
4                 Y

In [3]:
vine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5906333 entries, 0 to 5906332
Data columns (total 6 columns):
 #   Column             Dtype  
---  ------             -----  
 0   review_id          object 
 1   star_rating        float64
 2   helpful_votes      float64
 3   total_votes        float64
 4   vine               object 
 5   verified_purchase  object 
dtypes: float64(3), object(3)
memory usage: 270.4+ MB


# Filter to get 20 or more total votes 

In [4]:
filtered_vine_df = vine_df[vine_df['total_votes'] >= 20]
filtered_vine_df.head()


review_id  star_rating  helpful_votes  total_votes vine  \
211  R35PT06NWP7LDP          5.0           30.0         32.0    N   
344  R2P76PJFUGXBHO          1.0            6.0         45.0    N   
372   RQW4AFOG9MR4Z          5.0           51.0         52.0    N   
406  R2SMUEBMGLAJQK          5.0           29.0         36.0    N   
408  R1XVIZZALU5P6J          5.0          148.0        150.0    N   

    verified_purchase  
211                 N  
344                 N  
372                 N  
406                 Y  
408                 Y

# Find the most Helpful Votes

In [5]:
new_vine_df = filtered_vine_df.loc[filtered_vine_df.helpful_votes / filtered_vine_df.total_votes >= 0.5]
new_vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
211  R35PT06NWP7LDP          5.0           30.0         32.0    N   
372   RQW4AFOG9MR4Z          5.0           51.0         52.0    N   
406  R2SMUEBMGLAJQK          5.0           29.0         36.0    N   
408  R1XVIZZALU5P6J          5.0          148.0        150.0    N   
419  R11UFMG8M2488I          4.0           23.0         26.0    N   

    verified_purchase  
211                 N  
372                 N  
406                 Y  
408                 Y  
419                 N

# Vine Reviews

In [6]:
vine_reviews = new_vine_df[new_vine_df['vine'] == 'Y']
vine_reviews.head()

review_id  star_rating  helpful_votes  total_votes vine  \
50483   R6U9701C3BGO6          3.0          139.0        147.0    Y   
50625  R1XK3ALB45D7N4          5.0           33.0         34.0    Y   
50635  R1IZCSTLX81D6C          5.0           31.0         33.0    Y   
50656  R2C8NC8EQLH4JF          3.0           45.0         48.0    Y   
50890  R1JJ1YOJMOML1P          5.0           18.0         21.0    Y   

      verified_purchase  
50483                 N  
50625                 N  
50635                 N  
50656                 N  
50890                 N

# Reviews written / Not part of the Vine Program

In [8]:
not_vine_reviews = new_vine_df[new_vine_df['vine'] == 'N']
not_vine_reviews.head(10)

review_id  star_rating  helpful_votes  total_votes vine  \
211  R35PT06NWP7LDP          5.0           30.0         32.0    N   
372   RQW4AFOG9MR4Z          5.0           51.0         52.0    N   
406  R2SMUEBMGLAJQK          5.0           29.0         36.0    N   
408  R1XVIZZALU5P6J          5.0          148.0        150.0    N   
419  R11UFMG8M2488I          4.0           23.0         26.0    N   
548  R2OSHKSPXU68W5          5.0          136.0        147.0    N   
595  R3M02FSD3BLUPU          5.0           19.0         21.0    N   
701  R1P0LEEJHH09L3          5.0           32.0         32.0    N   
749  R1V6UU1EHW3Q12          5.0           57.0         62.0    N   
823  R230W87COFSTCL          5.0           21.0         21.0    N   

    verified_purchase  
211                 N  
372                 N  
406                 Y  
408                 Y  
419                 N  
548                 N  
595                 N  
701                 Y  
749                 Y  
823                 Y

# Five Star Review Analysis

In [9]:
count_vine = len(vine_reviews)
count_vine

33

#### Of all the reviews, 33 paid- vine reviews and 45,388 unpaid reviews

In [10]:
count_nonvine = len(not_vine_reviews)
count_nonvine

45388

In [11]:
max_rating_vine = len(vine_reviews[vine_reviews['star_rating']==5])
max_rating_vine


15

In [12]:
max_rating_nonvine = len(not_vine_reviews[not_vine_reviews['star_rating']==5])
max_rating_nonvine

23733

#### Of all the reviews, 15 vine reviews were 5 star, while 23,733 were non-paid 5star reviews.

# Five Star Review % by Program

In [13]:
max_rating_vine/count_vine

0.45454545454545453

In [14]:
max_rating_nonvine/count_nonvine

0.5228915131752886

#### Paid Vine Program

In [15]:
round(vine_reviews['star_rating'].mean(),2)

4.09

#### Unpaid Program 

In [16]:
round(not_vine_reviews['star_rating'].mean(),2)

3.87